public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  - [Set form model data using _setValue_ and _patchValue_](#set-data)
+  - [Use _FormArray_ to present an array of _FormGroups_](#form-array)
+  - [Watch control changes](#watch-control)
+  - [Save form data](#save)
+
+  Try the <live-example plnkr="final" title="Reactive Forms (final) in Plunker">Reactive Forms live-example</live-example>.
+
+  You can also run the <live-example title="Reactive Forms Demo in Plunker">Reactive Forms Demo</live-example> version
+  and choose one of the intermediate steps from the "demo picker" at the top.
+
+a#intro
+:marked
+  ## Introduction to Reactive Forms
+
+  Angular offers two form-building technologies: _reactive_ forms and _template-driven_ forms.
+  The two technologies belong to the `@angular/forms` library
+  and share a common set of form control classes.
+
+  But they diverge markedly in philosophy, programming style, and technique.
+  They even have their own modules: the `ReactiveFormsModule` and the `FormsModule`.
+
+  ### _Reactive_ forms
+  Angular _reactive_ forms facilitate a _reactive style_ of programming 
+  that favors explicit management of the data flowing between
+  a non-UI _data model_ (typically retrieved from a server) and a 
+  UI-oriented _form model_ that retains the states 
+  and values of the HTML controls on screen.
+
+  With _reactive_ forms, you create a tree of Angular form controls in code, and 
+  bind them to HTML widgets in the component template, using techniques described in this guide.
+
+  The component class is responsible for all data flowing between the component and the screen.
+  It has immediate access to both the data model and the form control structure. 
+  It pushes data model values into the form controls and 
+  pulls user-changed values back out of the form controls.
+  The component can _observe_ changes in form control state and _react_ to those changes.
+
+  In keeping with the reactive paradigm, the component should preserve the immutability of the _data model_,
+  treating it as a pure source of original values.
+  Rather than update the data model directly, 
+  the component extracts user changes and forwards them to an external component or service,
+  which does something with them (such as saving them) 
+  and returns a new _data model_ to the component that reflects the updated model state.
+
+  ### _Template-driven_ forms
+
+  _Template-driven_ forms, introduced in the [Template guide](forms.html), take a completely different approach.
+  
+  You place HTML form controls (such as `<input>` and `<select>`) in the component template and 
+  bind them to _data model_ properties in the component, using the `[(ngModel)]` two-way data binding directive.
+  
+  You don't create Angular form controls. Angular infers them from the data bindings.
+  You don't push and pull data values. Angular handles that for you with `[(ngModel)]`.
+  Angular updates the mutable _data model_ with user changes as they happen.
```

Comment:
@wardbell This pattern has been in all the Angular forms talks I've done (although I've only done two). Also I don't follow the circular logic that "it shouldn't be documented because you can't find documentation anywhere about it".  

Again, I'll stress that for simple forms, such concerns are unnecessary. You don't always have complicated scenarios where you need to heavy transformations. This pattern is perfectly valid.